### ISSUE TITLE
Latest quote price, error in sample code - "first argument must be a named list"

### ISSUE THREAD

**Issue by danielkhalilov** [yogat3ch/AlpacaforR#30]
Hello,

I am trying to pull the latest price of a symbol. I just merged jagg19 with yogat3ch. To pull the latest price, i was using: 
last_price <- get_poly_last_price("AMZN")$last$price

Now the documentation says that code is deprecated and throws a warning, though it still works. I noticed the following code in the order_submit documentation is meant to replace that: 
(.lq <- polygon("lq", symbol = "AMZN"))

When running it, I get the error: 
Error in list2env(.x, parent = .envir) : first argument must be a named list

Any help in resolving this error is much appreciated.

**Comment by yogat3ch** [comment]
@danielkhalilov
Since you closed this, I'm assuming you figured it out? 
What solved it?

**Comment by danielkhalilov** [comment]
@yogat3ch 
I actually was not able to solve this issue, so it may still be a known bug. I tried passing in "AMZN" as a list, but had no luck. I can reopen the ticket if you'd like.

The reason I closed the ticket is because I realized I do not need the yogat3ch library functionality in this algo (since "take profit at" oto does not go into after-hours, i decided to use a simple limit sell instead; the bracket order code does work as intended though, as it is not meant to go into after hours!). Therefore I un-merged the library and decided to stick with jagg19's get_poly_last_price, as this code was already set up for it.

Thanks for following up! Those bracket orders are very powerful, lot of good stuff in this library.

**Comment by yogat3ch** [comment]
You can leave it closed since it's on this repo. I'll look into the issue
and get back to you. Can you give me the output from `sessionInfo()`
though? It will help me to debug the issue.

--
Sent from a mobile device

Stephen Holsenbeck RYT-500
Graduate Certificate in Data Analytics '18
Master in Health Data Analytics - Dec '19

**Comment by danielkhalilov** [comment]
Sure, no problem! (edit: Updated quantmod from 0.4.15 to 0.4.17 since issue)

R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.15.4

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] AlpacaforR_0.1.0.9000      cronR_0.4.0                e1071_1.7-2                dplyr_0.8.3                jsonlite_1.7.0            
 [6] httr_1.4.1                 rmutil_1.1.3               candlesticks_0.2-8         tidyquant_0.5.9            PerformanceAnalytics_1.5.3
[11] lubridate_1.7.4            quantmod_0.4.17            xts_0.11-2                 zoo_1.8-6                  TTR_0.23-6                
[16] randomForest_4.6-14        plyr_1.8.4                 data.table_1.12.2          caret_6.0-84               ggplot2_3.2.1             
[21] lattice_0.20-38           

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1         class_7.3-14       assertthat_0.2.1   digest_0.6.23      ipred_0.9-9        foreach_1.4.4      R6_2.4.0          
 [8] stats4_3.5.1       pillar_1.4.1       rlang_0.4.7        lazyeval_0.2.2     curl_4.2           rstudioapi_0.11    rpart_4.1-13      
[15] Matrix_1.2-17      splines_3.5.1      gower_0.2.1        stringr_1.3.1      munsell_0.5.0      compiler_3.5.1     xfun_0.11         
[22] pkgconfig_2.0.2    nnet_7.3-12        tidyselect_0.2.5   tibble_2.1.3       prodlim_2018.04.18 quadprog_1.5-7     codetools_0.2-15  
[29] fansi_0.4.0        crayon_1.3.4       withr_2.1.2        MASS_7.3-51.4      recipes_0.1.5      ModelMetrics_1.2.2 grid_3.5.1        
[36] Quandl_2.10.0      nlme_3.1-142       gtable_0.3.0       magrittr_1.5       scales_1.0.0       cli_2.0.2          stringi_1.2.4     
[43] reshape2_1.4.3     timeDate_3043.102  generics_0.0.2     lava_1.6.6         iterators_1.0.10   tools_3.5.1        glue_1.3.1        
[50] purrr_0.3.4        survival_3.1-7     colorspace_1.4-1   knitr_1.26

**Comment by yogat3ch** [comment]
Thank you for that! 
You ran the following:
`(.lq <- polygon("lq", symbol = "AMZN"))`
which produced the error?
I'm unable to reproduce that error with AlpacaforR 1.0. 


Update 2020-07-28 08:31: Looking more closely at your `sessionInfo` It looks like you're not actually running AlpacaforR 1.0 found [here](https://github.com/yogat3ch/AlpacaforR). If you merged some of the pull requests on this repo, you're likely to encounter some bugs. The full version is at the link above.

**Comment by danielkhalilov** [comment]
Yes that is correct, i ran that line of code straight from the documentation.

If this is helpful, I get a similar error when I run list2env("AMZN")
Error in list2env("AMZN") : first argument must be a named list

I think you're right, given 1.0 is not backwards-compatible, i likely should not have been able to use any of the [deprecated] legacy functions from before that patch after integrating the yogat3ch merge, such as get_poly_last_price(). I probably should have updated to 1.0, then merged. It was my first time merging, so lesson learned lol - always update first! I'll update this thread if/once I update to the latest version and confirm. Thanks again for your hard work!

**Comment by yogat3ch** [comment]
The error is actually caused by some black magic stuff that I included to implicitly pass variables between internal function environments without having to pass them explicitly each time internals are called. It's been improved/debugged significantly since I made those pull requests.

> likely should not have been able to use any of the [deprecated] legacy functions from before that patch after integrating the yogat3ch merge, such as get_poly_last_price().

I did actually leave in as many deprecated functions in as I could, with a warning message - in effect *soft-deprecating* them. I'd like to preserve backward compatibility for folks who want to try out the new features and gradually get adjusted to using the new syntax and functions without having code break. However, there are some instances where I was not able to do so. 

>I'll update this thread if/once I update to the latest version and confirm. Thanks again for your hard work!

Cool, I think you'll find that the new version maps into the Alpaca API documentation in a very logical way. Its also quite a bit more user-friendly in terms of documentation and code verbosity (it generally takes far fewer characters to do the same thing in the new version). There's also substantially better error catching and informative messages for common user errors with arguments.

Enjoy!
